envoy/router/path_rewrite.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include "envoy/config/typed_config.h"
+#include "envoy/router/path_match.h"
+
+#include "source/common/common/logger.h"
+
+#include "absl/status/statusor.h"
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Router {
+
+/**
+ * Creates the new route path based on the provided rewrite pattern.
+ * Subclassing Logger::Loggable so that implementations can log details.
+ */
+class PathRewriter : Logger::Loggable<Logger::Id::router> {
+public:
+  PathRewriter() = default;
+  virtual ~PathRewriter() = default;
+
+  /**
+   * Used to determine if the matcher policy is compatible.
+   *
+   * @param path_match_policy current path match policy for route
+   * @param active_policy true if user provided policy
+   * @return true if current path match policy is acceptable
+   */
+  virtual absl::Status isCompatibleMatchPolicy(PathMatcherSharedPtr path_match_policy,
+                                               bool active_policy) const PURE;
+
+  /**
+   * Used to rewrite the current url to the specified output. Can return a failure in case rewrite
```

Comment:
Here and elsewhere please update function comments to match the style guide. In particular:

> Function comments should be written with an implied subject of _This function_ and should start with the verb phrase; for example, "Opens the file", rather than "Open the file".

https://google.github.io/styleguide/cppguide.html#Function_Comments